app/dose-amigos-user-service/dose-amigos-user.service.ts

```diff
@@ -1,20 +1,32 @@
 import {Injectable} from "@angular/core";
 import {DoseAmigosUser} from "../dose-amigos-user/dose-amigos-user";
-import {DOSE_AMIGOS_USERS} from "./dose-amigos-user-mocks";
+import {BACKEND_URL} from "../backend-config/backend-config";
+import {AuthHttp} from "../angular2-jwt";
 
 /**
  * Service for fetching and saving DoseAmigosUser instances.
  */
 @Injectable()
 export class DoseAmigosUserService {
 
+    private amigosUsersUrl: string = `${BACKEND_URL}/amigo-users`;
+
+    constructor(
+        private http: AuthHttp
+    ) {
+    }
+
     /**
      * Get a list of DoseAmigosUsers.
      * @returns {Promise<Array<DoseAmigosUser>>}
      */
     public list(): Promise<Array<DoseAmigosUser>> {
-        return Promise.resolve(
-            DOSE_AMIGOS_USERS
+        return this.http.get(
+            this.amigosUsersUrl
+        ).toPromise().then(
+            (response) => response.json()
+        ).catch(
+            this.handleError
         );
     }
```

Comment:
This is the only method used by client-side code today.

Used in the Amigos listing page.